internal/keppel/config.go

```diff
@@ -139,3 +148,51 @@ func mayGetenvURL(key string) *url.URL {
 	}
 	return parsed
 }
+
+//GetenvOrDefault is like os.Getenv but it also takes a default value which is
+//returned if the given environment variable is missing or empty.
+func GetenvOrDefault(key, defaultVal string) string {
+	val := os.Getenv(key)
+	if val == "" {
+		val = defaultVal
+	}
+	return val
+}
+
+func getDbURL() url.URL {
+	dbName := GetenvOrDefault("KEPPEL_DB_NAME", "keppel")
+	dbUsername := GetenvOrDefault("KEPPEL_DB_USERNAME", "postgres")
+	dbPass := os.Getenv("KEPPEL_DB_PASSWORD")
+	dbHost := GetenvOrDefault("KEPPEL_DB_HOSTNAME", "localhost")
+	dbPort := GetenvOrDefault("KEPPEL_DB_PORT", "5432")
+	dbConnOpts := os.Getenv("KEPPEL_DB_CONNECTION_OPTIONS")
+
+	return url.URL{
+		Scheme:   "postgres",
+		User:     url.UserPassword(dbUsername, dbPass),
+		Host:     net.JoinHostPort(dbHost, dbPort),
+		Path:     dbName,
+		RawQuery: dbConnOpts,
+	}
+}
+
+// GetRedisURL returns a Redis connection URL by getting the required
+// parameters from environment variables:
+//   REDIS_PASSWORD, REDIS_HOSTNAME, REDIS_PORT, and REDIS_DB_NUM.
+//
+// The environment variable keys are prefixed with the provided prefix.
+func GetRedisURL(prefix string) string {
+	prefix = strings.Join([]string{prefix, "REDIS"}, "_")
+	pass := os.Getenv(prefix + "_PASSWORD")
+	host := GetenvOrDefault(prefix+"_HOSTNAME", "localhost")
+	port := GetenvOrDefault(prefix+"_PORT", "6379")
+	dbNum := GetenvOrDefault(prefix+"_DB_NUM", "0")
+
+	url := url.URL{
+		Scheme: "redis",
+		User:   url.UserPassword("", pass),
+		Host:   net.JoinHostPort(host, port),
+		Path:   dbNum,
+	}
+	return url.String()
+}
```

Comment:
Instead of generating a URL that we then parse with `redis.ParseURL`, this should return a `*redis.Options` directly.